cmd/tusd/cli/hooks/grpc.go

```diff
@@ -0,0 +1,111 @@
+package hooks
+
+import (
+	"context"
+	"io"
+	"log"
+	"time"
+
+	"github.com/golang/protobuf/proto"
+	grpc_retry "github.com/grpc-ecosystem/go-grpc-middleware/retry"
+	"github.com/tus/tusd/pkg/handler"
+	pb "github.com/tus/tusd/pkg/proto/v1"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/status"
+)
+
+type GrpcHook struct {
+	Endpoint   string
+	MaxRetries int
+	Backoff    int
+}
+
+func (_ GrpcHook) Setup() error {
+	return nil
+}
+
+func (g GrpcHook) InvokeHook(typ HookType, info handler.HookEvent, captureOutput bool) ([]byte, int, error) {
+	ctx := context.Background()
+	opts := []grpc_retry.CallOption{
+		grpc_retry.WithBackoff(grpc_retry.BackoffLinear(time.Duration(g.Backoff) * time.Second)),
+		grpc_retry.WithMax(uint(g.MaxRetries)),
+	}
+	conn, err := grpc.Dial(g.Endpoint,
+		grpc.WithInsecure(),
+		grpc.WithStreamInterceptor(grpc_retry.StreamClientInterceptor(opts...)),
+		grpc.WithUnaryInterceptor(grpc_retry.UnaryClientInterceptor(opts...)),
+	)
+	if err != nil {
+		return nil, 0, err
+	}
+	defer conn.Close()
+	client := pb.NewHookServiceClient(conn)
+	stream, err := client.Send(ctx)
+	if err != nil {
+		if e, ok := status.FromError(err); ok {
+			return nil, int(e.Code()), err
+		}
+		return nil, 0, err
+	}
```

Comment:
I wonder if this part should go into `Setup` instead. I'm just guessing here, but this looks like a one time setup.

Reply:
Right, I moved gRPC options to Setup